### PR TITLE
feat: Revised PDP - About Participant (M2-7767)

### PR DESCRIPTION
**Pull request by farmerpaul**
- [ ] ~~Tests for the changes have been added~~ **(moved to separate task)**
- [ ] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

🔗 [Jira Ticket M2-7767](https://mindlogger.atlassian.net/browse/M2-7767)

This PR introduces the new tabbed view of the Participant Details Page (PDP), and specifically the first tab of that screen, "About Participant". Changes include:

- Create navigation infrastructure to display this new tabbed view when the `enableActivityAssign` feature flag is enabled, and the original PDP when disabled
- Add support for new endpoint `/activities/applet/${appletId}/target/${subjectId}`
- Flesh out "About Participant" view, with a reusable hook, `useAssignmentsTab`, that can be reused by the "By Participant" view (assign, unassign, navigate to data, export data, unlock applet)
- Add empty state to About Participant view
- Slightly refactor `useAsync` to use a new, less awkward pattern for preserving the previous value when data is refetched (see [this commit message](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1936/commits/999927b33d400341b443a46f333302b90d960ab7) for details)

> [!TIP]
> Related changes have been grouped by commit, with commit messages containing more detailed descriptions as needed. Reviewing [commit-by-commit](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1936/commits) may be easier.

### 📸 Screenshots

https://github.com/user-attachments/assets/b37c0458-95f4-4761-9c30-0df6b8c1ca2f

### 🪤 Peer Testing

> [!NOTE] 
> This feature depends on the `enableActivityAssign` feature flag. This should already be enabled in dev/local environments. To control its value manually, add `features.enableActivityAssign = [true|false]` prior to the `return` statement [here in `useFeatureFlags`](https://github.com/ChildMindInstitute/mindlogger-admin/blob/e6ff173fc6aaf31da3e81d92ab4f59caa8eef6da/src/shared/hooks/useFeatureFlags.ts#L25-L26).

#### Test with feature flag enabled ✅

1. Add `features.enableActivityAssign = true` to [`useFeatureFlags`](https://github.com/ChildMindInstitute/mindlogger-admin/blob/e6ff173fc6aaf31da3e81d92ab4f59caa8eef6da/src/shared/hooks/useFeatureFlags.ts#L25-L26).
2. Please refer to the AC of [M2-7767](https://mindlogger.atlassian.net/browse/M2-7767) to ensure behaviour is in alignment.

#### Test with feature flag disabled 🚫

1. Add `features.enableActivityAssign = false` to [`useFeatureFlags`](https://github.com/ChildMindInstitute/mindlogger-admin/blob/e6ff173fc6aaf31da3e81d92ab4f59caa8eef6da/src/shared/hooks/useFeatureFlags.ts#L25-L26).
2. Load the PDP and ensure there are no regressions in behaviour.

[M2-7767]: https://mindlogger.atlassian.net/browse/M2-7767?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/resources/app-fr.json [hunk 1]
```diff
@@ -276,6 +276,12 @@
   "assign": "Attribuer",
   "assignActivity": "Attribuer l'activité",
   "assignFlow": "Attribuer le flux",
+  "assignActivityDisabledTooltip": "Cette activité est définie sur « Caché » dans le Créateur d'Activités et ne peut donc pas être attribuée ici.",
+  "assignFlowDisabledTooltip": "Ce flux est défini sur « Caché » dans le Créateur de Flux d'Activités et ne peut donc pas être attribué ici.",
+  "assignToTeamMemberTooltip": "Le participant actuel est un membre de l'équipe et ne peut donc pas être assigné en tant que sujet d'une activité ou d'un flux.",
+  "assignments": "Attributions",
+  "activitiesByParticipant": "Par le participant",
+  "activitiesAboutParticipant": "Concernant le participant",
```
Comment (sultanofcardio) [inline src/resources/app-fr.json]:
```suggestion
```

File: src/modules/Dashboard/features/Participant/Assignments/ByParticipant/ByParticipant.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import { AssignmentsTab } from '../AssignmentsTab';
+
+const ByParticipant = () => {
+  const { t } = useTranslation();
+
+  return <AssignmentsTab />;
+};
+
+export default ByParticipant;
```
Comment (sultanofcardio) [inline src/modules/Dashboard/features/Participant/Assignments/ByParticipant/ByParticipant.tsx]: Personally I think this is fine, but it seems to introduce an ESLint warning

```suggestion
import { AssignmentsTab } from '../AssignmentsTab';

const ByParticipant = () => <AssignmentsTab />;

export default ByParticipant;

```

**Review by sultanofcardio — APPROVED** [review]
lgtm

File: src/shared/hooks/useAsync/useAsync.ts [hunk 3]
```diff
@@ -1,38 +1,51 @@
...
 
+import {
+  ErrorCallback,
+  FinallyCallback,
+  optionsIsObjectTypeGuard,
+  SuccessCallback,
+  UseAsyncOptions,
+} from './useAsync.types';
+
 export const useAsync = <T, K>(
```
Comment (mbanting) [inline src/shared/hooks/useAsync/useAsync.ts:31]: `useAsync` is a highly used hook used in almost 100 places. Might be a good opportunity to include a couple of lines of documentation. The options are straightforward except for `retainValue` which might be mysterious for some new to the codebase.
```suggestion
/**
 * Custom hook to handle asynchronous operations with callbacks,
 * and a retainValue flag to keep the previous value during the loading state.
 */
export const useAsync = <T, K>(
```

**Review by mbanting — APPROVED** [review]
Another clean and well organized PR @farmerpaul 🚀 
Couldn't see anything to improve, just one small suggestion.

**Pull request merged**